transport-native-unix-common/src/main/java/io/netty/channel/unix/Socket.java

```
@@ -257,6 +268,18 @@ public final boolean finishConnect() throws IOException {
         return true;
     }
 
+    public final boolean disconnect() throws IOException {
+        int res = disconnect(fd);
+        if (res < 0) {
+            if (res == ERRNO_EINPROGRESS_NEGATIVE) {
```

@Scottmitch I am not sure this will ever happen. I guess we may want to just throw an exception i this case. WDYT ?